arch/cortex-m4/src/mpu.rs

```diff
@@ -163,8 +163,18 @@ impl kernel::mpu::MPU for MPU {
         } else {
             // Memory base not aligned to memory size
 
-            // Which subregion size would align with the base address?
-            let subregion_size = start % len;
+            // Which (power-of-two) subregion size would align with the base
+            // address?
+            //
+            // We find this by taking smallest binary substring of the base
+            // address with exactly one bit:
+            //
+            //      1 << (start.trailing_zeros())
+            //
+            // The zero case (where there are no 1 bits) is taken care of with
+            // the `wrapping_shl` operation, which masks bits that would
+            // overflow the integer width.
+            let subregion_size = (1 as usize).wrapping_shl(start.trailing_zeros());
```

Comment:
In what scenario would we encounter the need for the `wrapping_shl`? If I understand correctly, it only applies when `start == 0` (as `0.trailing_zeros()` == 32, and `(anything greater than zero).trailing_zeros() < 32`, however, if start is 0, then `0 % len` for any value of len will be 0, and we'll be in the above if branch